scripts/babel/wrap-warning-with-env-check.js

```diff
@@ -25,23 +25,35 @@ module.exports = function(babel, options) {
           }
 
           if (path.get('callee').isIdentifier({name: 'warning'})) {
-            node[SEEN_SYMBOL] = true;
-
             // Turns this code:
             //
             // warning(condition, argument, argument);
             //
             // into this:
             //
             // if (__DEV__) {
-            //   warning(condition, argument, argument);
+            //   if (!condition) {
+            //     warning(false, argument, argument);
+            //   }
             // }
             //
-            // The goal is to strip out warning calls entirely in production.
+            // The goal is to strip out warning calls entirely in production
+            // and to avoid evaluating the arguments in development.
+            const condition = node.arguments[0];
+            const newNode = t.callExpression(
+              node.callee,
+              [t.booleanLiteral(false)].concat(node.arguments.slice(1))
+            );
+            newNode[SEEN_SYMBOL] = true;
```

Comment:
Seems kind of silly that we're using fbjs for `warning` and `invariant` at this point 😁

Reply:
I think we still need for React Native (and to go through our custom fbsource RN infra there).
Although we might as well have implemented that through the new forking infra.

Reply:
Right, that's what I meant. We may have well just built our own thing.